### PR TITLE
[Dependent][CVAT-M2] Oracles M2 API - endpoints

### PR DESCRIPTION
**Pull request by zhiltsov-max**
## Description

Changes for new Human APP version.

## Summary of changes

Depends on #2094

- Upgraded pydantic to v2 and fastapi to 0.111.1, migrated code
- Updated ExcOr endpoints (GET /job, POST | GET /assignment, PUT /register)
- Implemented filtering, pagination, JWT auth, throttling for the endpoints
- Added /stats and /stats/assignment endpoints, /assingments/{id}/resign

TBD:
- [x] Fix discrepancies between doc and SoW APIs
- [x] Rename "items" to "results" in list endpoints
- [x] Upgrade deps in RecOr
- [ ] Update tests

## Related issues
[Keywords for linking issues](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

### CODE REVIEW

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py [hunk 1]
```diff
@@ -1,120 +1,503 @@
...
+
+
+@router.get("/stats/assignment", description="Get oracle statistics for the user")
+async def get_user_stats(token: AuthorizationData = AuthorizationParam) -> UserStatsResponse:
+    wallet_address = token.wallet_address
+
+    with SessionLocal.begin() as session:
+        stats = {}
+
+        stats["assignments_amount"] = (
```
Comment (Dzeranov) [inline packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py:402]: `assignments_total`

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py [hunk 2]
```diff
@@ -1,120 +1,503 @@
...
+        )
+
+        stats["escrows_cancelled"] = (
+            session.query(cvat_service.Project.escrow_address)
+            .distinct()
+            .where(cvat_service.Project.status == ProjectStatuses.canceled)
+            .count()
+        )
+
+        stats["workers_amount"] = session.query(cvat_service.User.wallet_address).count()
```
Comment (Dzeranov) [inline packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py:483]: `workers_total`

**Review by Dzeranov — CHANGES_REQUESTED** [review]

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py (hunk 2 above)
Comment (zhiltsov-max) [reply packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py]: Fixed

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py (hunk 1 above)
Comment (zhiltsov-max) [reply packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py]: Fixed

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py [hunk 3]
```diff
@@ -1,120 +1,503 @@
...
+
+@router.get(
+    "/assignment",
+    description="Lists assignments",
+    response_model_exclude_unset=True,  # required for field selection
+    response_model_by_alias=True,  # required for pagination
+)
+async def list_assignments(
+    filter: AssignmentFilter = FilterDepends(AssignmentFilter),
+    created_after: Optional[datetime] = Query(default=None),
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py:253]: Added

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py [hunk 4]
```diff
@@ -1,120 +1,503 @@
...
-        elif email_db_user.wallet_address != user.wallet_address:
-            raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="User already exists")
+        elif email_db_user.wallet_address != user_wallet_address:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST, detail="User already exists"
+            )
 
         return UserResponse(
             wallet_address=email_db_user.wallet_address,
             cvat_email=email_db_user.cvat_email,
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py]: Changed

File: packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py [hunk 5]
```diff
@@ -1,120 +1,503 @@
...
+        def _page_serializer(
+            projects: Sequence[cvat_service.Project],
+        ) -> Sequence[JobResponse]:
+            page = [serialize_job(p, session=session) for p in projects]
+            return [filter.select_fields_(p) for p in page]
+
+        return paginate(session, query, transformer=_page_serializer)
+
+
+@router.put("/register", description="Binds a CVAT user to a HUMAN App user")
```
[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply packages/examples/cvat/exchange-oracle/src/endpoints/exchange.py]: Changed

**Review by Dzeranov — APPROVED** [review]

**Pull request merged**